### PR TITLE
integrated detekt

### PR DESCRIPTION
**Pull request by norangebit**
- update gradle wrapper to 4.10. Se more at https://github.com/arturbosch/detekt/issues/1195
- edit build file to integrate detekt
- add `config/detekt/detekt.yml` file with detekt base configuration
- `detekt` task is automatically run when executing `check`
- issue: #28

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,3 +55,27 @@ compileTestKotlin {
 task wrapper(type: Wrapper) {
     gradleVersion = '4.7'
 }
+
+detekt {
+    input = files("src/main/kotlin")
+
+    config = files("config/detekt/detekt.yml")
+
+    ignoreFailures = true // If`true` the build doesn't fail when the maxIssues count was reached.
```
Comment (Edvinas01) [inline build.gradle]: I'd leave this on. Currently there are 9 issues, the default for failure seems to be 10 so it passes. Someone in the future will have to address them at some point though :D

File: build.gradle [hunk 2]
```diff
@@ -54,3 +55,27 @@ compileTestKotlin {
 task wrapper(type: Wrapper) {
     gradleVersion = '4.7'
 }
+
+detekt {
+    input = files("src/main/kotlin")
+
+    config = files("config/detekt/detekt.yml")
```
Comment (Edvinas01) [inline build.gradle]: I'd remove `detekt.yml` as it mirrors the defaults. Personally I would use the config only for overrides via `buildUponDefaultConfig = true` flag - but I don't think that is necessary at the moment.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.7-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10-bin.zip
```
Comment (Edvinas01) [inline gradle/wrapper/gradle-wrapper.properties]: The `wrapper` task should be updated as well:
```groovy
task wrapper(type: Wrapper) {
    gradleVersion = '4.10'
}
```

File: build.gradle (hunk 1 above)
Comment (norangebit) [reply build.gradle]: So, do I set it to false?

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (norangebit) [reply gradle/wrapper/gradle-wrapper.properties]: You're right.

File: build.gradle (hunk 1 above)
Comment (Edvinas01) [reply build.gradle]: It defaults to false, I'd just remove it (https://arturbosch.github.io/detekt/groovydsl.html).

Comment (norangebit) [reply build.gradle]: I've already done like that.
I've also done the rebase of master, so you don't have merge conflicts.

**Pull request merged**